src/kpi/statistics/couplingOfComponents/lib/prFilesUtil.ts

```diff
@@ -0,0 +1,86 @@
+import { ObjectId } from 'mongoose';
+import { coupling } from '../model/coupling';
+
+/**
+ * It computes the coupling over all PRs and their changed files
+ * from @param prFiles coupling size of @param couplingSize and
+ * it considers occurences of the couplings of at least @param occs.
+ * @returns an array with all couplings, occurences and related PRs.
+ */
+export function getCoupling(
+  prFiles: { _id: ObjectId; changedFiles: string[] }[],
+  couplingSize: number = 3,
+  occs: number = 3,
+): coupling[] {
+  const combinationMap = new Map<
+    string,
+    { occs: number; pullrequests: string[] }
+  >();
+
+  prFiles.forEach((diff) => {
+    const combinations = binomialSubsets(diff.changedFiles, couplingSize);
+    const pr = diff._id.toString();
+    combinations.forEach((combo) => {
+      const key: string = combo.sort().join(' ');
+      if (combinationMap.has(key)) {
+        increaseCounterAndAddPR(key, pr);
+      } else {
+        addFileCombinationToMap(key, pr);
+      }
+    });
+  });
+
+  deleteOccurences(occs);
+
+  // sorting map entries descending by occurences
+  const res = [...combinationMap.entries()].sort(
+    (a, b) => b[1]['occs'] - a[1]['occs'],
+  );
+
+  return res;
+
+  function increaseCounterAndAddPR(key: string, pr: string) {
```

Comment:
pr: string is actually the pull request id I believe? 
please use clearer naming here

Reply:
Yes, you are right. I'll rename this.